neuralop/losses/scale_consistency_losses.py

```diff
@@ -0,0 +1,141 @@
+
+    Attributes
+    ----------
+        model : nn.Module
+            The neural operator.
+        x : torch.Tensor
+            The input tensor of shape (B, C, H, W) or (B, C, T, H, W).
+        loss_fn : callable
+            The loss function to compute the difference between the two evaluations (usually Lploss).
+        y : torch.Tensor, optional
+            The ground truth tensor of shape (B, C, H, W) or (B, C, T, H, W). If provided,
+            the loss is computed between model(x_small) and y_small.
+            If not provided, the loss is computed between model(x_small) and y_small where
+            y = model(x).
+        re : torch.Tensor, optional
+            Additional input tensor of shape (B, 1) if required by the model.
+        rate : float, optional
+            The rate at which to crop the input tensor. If None, a random rate is chosen.
+        new_y : torch.Tensor, optional
+            Not used in this function.
+        size_min : int, optional
+            The minimum size of the cropped subdomain.
+        type : str, optional
+            The type of PDE ('darcy', 'helmholtz', 'NS', 'burgers') to determine boundary extraction.
+        plot : bool, optional
+            If True, plots the cropped regions (not implemented here).
+        group_action : callable, optional
+            A function to apply group actions on (x_small, y_small) if needed.
+        align_corner : bool, optional
+            If True, aligns the corners of the cropped regions.
+
+    Returns:
+        torch.Tensor
+            The computed self-consistency loss.
+    """
+    #
+
+    batch_size = x.shape[0]
+
+    if type == "darcy":
+        ExtractBD = DarcyExtractBC
+        transform_xy = RandomCropResize(p=1.0, size_min=size_min)
+    elif type == "helmholtz":
+        ExtractBD = HelmholtzExtractBC
+        transform_xy = RandomCropResize(p=1.0, size_min=size_min)
+    elif type == "NS":
+        ExtractBD = lambda x, y: x  # No boundary
+        transform_xy = RandomCropResizeTimeAR(p=1.0, size_min=size_min)
+    elif type == "burgers":
+        ExtractBD = lambda x, y: BurgersExtractBC(y)
+        transform_xy = RandomCropResizeTime(p=1.0, size_min=size_min)
+    else:
+        print("boundary type not supported")
+
+    if re == None:
+        re = torch.ones(batch_size, 1, requires_grad=False).to(x.device)
+
+    # If y is given, we use it as the ground truth. the gradient only flow to y_small_
+    if y is not None:
+
+        # resample on smaller domain
+        i, j, h, w, re = transform_xy.get_params(x, y, re=re, rate=rate)
+
+        x_small = transform_xy.crop(x, i, j, h, w)
+        y_small = transform_xy.crop(y, i, j, h, w)
+        x_small = ExtractBD(x_small, y_small)
+
+        if group_action is not None:
+            x_small, y_small = group_action(x_small, y_small)
+
+        if type == "darcy":
+            y_small_ = model(x_small)
+        else:
+            y_small_ = model(x_small, re)
```

Comment:
This forward call seems like it might belong better in the custom trainer's training loop

Reply:
Yea we added a custom trainer, but I feel it will also be beneficial to have a loss, which we can use in other trainers in the future.